cluster-autoscaler/processors/binpacking/binpacking_limiter.go

```diff
@@ -25,11 +25,13 @@ import (
 // BinpackingLimiter processes expansion options to stop binpacking early.
 type BinpackingLimiter interface {
 	InitBinpacking(context *context.AutoscalingContext, nodeGroups []cloudprovider.NodeGroup)
-	StopBinpacking(context *context.AutoscalingContext, evaluatedOptions []expander.Option, usedNodeGroups map[string]bool) bool
+	StopBinpacking(context *context.AutoscalingContext, evaluatedOptions []expander.Option) bool
+	MarkProcessed(context *context.AutoscalingContext, nodegroupId string)
 }
 
 // NoOpBinpackingLimiter returns true without processing expansion options.
 type NoOpBinpackingLimiter struct {
+	processedNodeGroups map[string]bool
```

Comment:
Why do you need it here? It is only updated, never read. (Which makes sense if this is a no-op implementation.)

Reply:
Makes sense. Removed it from here.